Merge build assert macros

ozhuraki opened the pull request:
In order to de-duplicate 2 macros with the same use,
one of which is discouraged/deprecated, merge
BUILD_ASSERT(), BUILD_ASSERT_MSG() into one macro
and drop BUILD_ASSERT_MSG().

Previous PR: https://github.com/zephyrproject-rtos/zephyr/pull/23437

@carlescufi @utzig @nvlsianpu @anangl @nashif

If there's still an interest, let's proceed:

1. Upgrade mcuboot JuulLabs-OSS/mcuboot#682
2. Upgrade hal_nordic zephyrproject-rtos/hal_nordic#29
3. I could then drop the manifest changes in this PR.

ozhuraki committed:
global: Replace BUILD_ASSERT_MSG() with BUILD_ASSERT()

Replace all occurences of BUILD_ASSERT_MSG() with BUILD_ASSERT()
as a result of merging BUILD_ASSERT() and BUILD_ASSERT_MSG().

Signed-off-by: Oleg Zhurakivskyy <oleg.zhurakivskyy@intel.com>

ozhuraki committed:
toolchain: common: Merge build assert macros

In order to de-duplicate 2 macros with the same use,
merge BUILD_ASSERT(), BUILD_ASSERT_MSG() into one macro.

Make BUILD_ASSERT_MSG() deprecated.

Signed-off-by: Oleg Zhurakivskyy <oleg.zhurakivskyy@intel.com>

ozhuraki committed:
drivers: ieee802154: Add a missing include

Add a missing include of debug/stack.h in order to fix the
compilation warning on implicit declaration of log_stack_usage().

Signed-off-by: Oleg Zhurakivskyy <oleg.zhurakivskyy@intel.com>

carlescufi commented on west.yml:
```diff
@@ -47,7 +47,7 @@ manifest:
       path: modules/hal/cypress
     - name: hal_nordic
-      revision: f6d5eca68e07551d6a36359b67600f9e98736839
+      revision: pull/29/head
```
you can use the SHA here now

ozhuraki replied:
@carlescufi 

OK, thanks. Done, dropped manifest changes for hal_nordic.

carlescufi replied:
@ozhuraki I meant the new SHA, see comment by @nvlsianpu

carlescufi commented on west.yml:
```diff
@@ -77,7 +77,8 @@ manifest:
     - name: mcuboot
-      revision: 470e2f3d5bdcb1892d76a534a745186fd8ff5aab
+      url: https://github.com/JuulLabs-OSS/mcuboot.git
+      revision: pull/682/head
```
@nvlsianpu will you upmerge our copy of MCUboot?

nvlsianpu reviewed (changes requested):
hal_nordic module is not upgraded
@ozhuraki : 59d33bc1dd59df14feb26f0010d527c1ea7a89e8

nvlsianpu commented:
I had to reverted your mcuboot patch as it break CI for other PR we needed.
Your patch is replayed here: https://github.com/JuulLabs-OSS/mcuboot/pull/692
Please point it in manifest as well.

ozhuraki committed:
west: Point hal_nordic to the BUILD_ASSERT() patch

Temporary point hal_nordic to the revision
59d33bc1dd59df14feb26f0010d527c1ea7a89e8
where the changes are present (before it's updgraded),
so the CI can proceed.

Signed-off-by: Oleg Zhurakivskyy <oleg.zhurakivskyy@intel.com>

ozhuraki commented:
@nvlsianpu 

> hal_nordic module is not upgraded 59d33bc1dd59df14feb26f0010d527c1ea7a89e8

Done.

> JuulLabs-OSS/mcuboot#692

Done.

nvlsianpu commented:
@carlescufi Can we merge this today and end the story?

nvlsianpu commented:
^^ If yes I will updated zphyr-rtos/mcuboot and manifest here

nvlsianpu commented:
DNM untill mcuboot repo is not synchronized.

nvlsianpu committed:
manifest: upgrade mcuboot

Include build assert macros rework.
Update pip requirements by cbor package required by MCUBoot since
https://github.com/JuulLabs-OSS/mcuboot/pull/674 was merge.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu commented:
need to rebase after https://github.com/zephyrproject-rtos/zephyr/pull/23552 will be merged.

nvlsianpu commented on scripts/requirements-run-test.txt:
```diff
@@ -10,3 +10,6 @@ pyocd>=0.24.0
 tabulate
+
+# used by mcuboot
+'cbor>=1.0.0'
```
@mbolivar-nordic Is this OK?
cbor package is required by MCUBoot since https://github.com/JuulLabs-OSS/mcuboot/pull/674 was merge

nvlsianpu replied:
typo xD

nvlsianpu commented:
```console
File "/home/buildslave/src/github.com/zephyrproject-rtos/bootloader/mcuboot/scripts/imgtool/boot_record.py", line 17, in <module>
    import cbor
ModuleNotFoundError: No module named 'cbor'
```

^^ cbor package is required by MCUBoot since JuulLabs-OSS/mcuboot#674 was merge
I added `cbor`  to requirements, but this was not enough to force CI environment to install it.
How this should be enforced?

@carlescufi, @nvlsianpu  Can you help with this.

carlescufi commented:
> ```
> File "/home/buildslave/src/github.com/zephyrproject-rtos/bootloader/mcuboot/scripts/imgtool/boot_record.py", line 17, in <module>
>     import cbor
> ModuleNotFoundError: No module named 'cbor'
> ```
> 
> ^^ cbor package is required by MCUBoot since [JuulLabs-OSS/mcuboot#674](https://github.com/JuulLabs-OSS/mcuboot/pull/674) was merge
> I added `cbor` to requirements, but this was not enough to force CI environment to install it.
> How this should be enforced?
> 
> @carlescufi, @nvlsianpu Can you help with this.

@galak do you know why CI is not picking up this commit: https://github.com/zephyrproject-rtos/zephyr/pull/23625/commits/d701d27559813aebe5e7ae8db35020396cd142bb
could it be because it's in requirements-extras ?

nvlsianpu commented:
^^ it is in requirements-run-test.txt

galak commented:
> @galak do you know why CI is not picking up this commit: [d701d27](https://github.com/zephyrproject-rtos/zephyr/commit/d701d27559813aebe5e7ae8db35020396cd142bb)
> could it be because it's in requirements-run-test?

Its because we don't do pip install as part of the number CI process.  We expect the docker container to already have those things installed.

carlescufi commented:
> > @galak do you know why CI is not picking up this commit: [d701d27](https://github.com/zephyrproject-rtos/zephyr/commit/d701d27559813aebe5e7ae8db35020396cd142bb)
> > could it be because it's in requirements-run-test?
> 
> Its because we don't do pip install as part of the build CI process. We expect the docker container to already have those things installed.

@galak so that means that we freeze the packages to whatever is on Zephyr's and MCUBoot master branches at the time we generate the image, or do we build the image every time we run CI?

See here:

https://github.com/zephyrproject-rtos/ci-dockerfiles/blob/master/shippable/install.sh#L135
https://github.com/zephyrproject-rtos/ci-dockerfiles/blob/master/shippable/install.sh#L143

galak commented:
> @galak so that means that we freeze the packages to whatever is on Zephyr's and MCUBoot master branches at the time we generate the image, or do we build the image every time we run CI?
> 
> See here:
> 
> https://github.com/zephyrproject-rtos/ci-dockerfiles/blob/master/shippable/install.sh#L135
> https://github.com/zephyrproject-rtos/ci-dockerfiles/blob/master/shippable/install.sh#L143

We build the image when we need it, so updates to requirements would mean someone needs to rebuild the CI image.

nvlsianpu commented:
> We build the image when we need it, so updates to requirements would mean someone needs to rebuild the CI image.

@galak Who can do that?

galak commented:
> @galak Who can do that?

I kicked off a v0.11.6 release of the ci-dockerfiles so once that completes that should have the cbor python package and we can update .shippable.yml to point to it.

carlescufi committed:
ci: docker: Update image to 0.11.6

This includes the cbor Python package required by MCUBoot.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@ozhuraki @nvlsianpu I have rebased, fixed conflicts and updated the docker image

ozhuraki commented:
@nvlsianpu @anangl @carlescufi @galak 

Thanks!
